File: solutions/3705-find-the-largest-almost-missing-integer/find-the-largest-almost-missing-integer.java
class Solution {
    public int largestInteger(int[] nums, int k) {
        if (nums.length < k) {
            return -1;
        }

        Map<Integer, Integer> hm = new HashMap<>();

        int i = 0;
        int j = i + k - 1;

        while (j < nums.length) {
            Set<Integer> hs = new HashSet<>();
            for (int x = i; x <= j; x++) {
                hs.add(nums[x]);
            }

            for (int num : hs) {
                hm.put(num, hm.getOrDefault(num, 0) + 1);
            }
            i++;
            j++;
        }
        int max_val = -1;
        for (Map.Entry<Integer, Integer> e : hm.entrySet()) {
            if (e.getValue() == 1) {
                max_val = Math.max(max_val, e.getKey());
            }
        }

        return max_val;
    }
}

File: solutions/3705-find-the-largest-almost-missing-integer/find-the-largest-almost-missing-integer.rs
impl Solution {
    pub fn largest_integer(nums: Vec<i32>, k: i32) -> i32 {
        let n = nums.len();
        let k = k as usize;
        if n == k {
            return *nums.iter().max().unwrap();
        }
        let mut count = [0; 51];
        for &x in &nums {
            count[x as usize] += 1;
        }
        if k == 1 {
            for i in (0..=50).rev() {
                if count[i as usize] == 1 {
                    return i;
                }
            }
            return -1;
        }
        let mut res = -1;
        if count[nums[0] as usize] == 1 {
            res = res.max(nums[0]);
        }
        if count[nums[n - 1] as usize] == 1 {
            res = res.max(nums[n - 1]);
        }
        res
    }
}